### DISCUSSION TITLE
Migration Guide for 3.0

### DISCUSSION THREAD

**Discussion by ocelot2123** [saleor/saleor#7868] [Q&A] [answered]
Hi, we are currently running Saleor 2.10.1 and are doing a major refactoring so we figured we might as well try to upgrade to 3.0 as it seems like it is nearing launch and we could use some of its features. Just wondering if there will be a detailed step by step migration guide for old versions? If so, when can we expect that to come out?

**Comment by karolkielecki** [comment]
@ocelot2123 Hey! We'll provide a migration guide. You can also find some more info [here](https://github.com/mirumee/saleor/discussions/7115#discussioncomment-577346).

**Comment by ocelot2123** [comment]
so if we currently have a 2.10.1 project with some custom code and is looking to migrate to 3.0. We are thinking of doing it in 2 ways. 1 is to try to do a merge from a 3.0 branch, second is to fork the 3.0 project and add back our custom changes. In the other post you linked it said "data will be migrated automatically". I'm not quite sure what that means or what the migration process would look like, so if you could give a bit more information on that so we can decide our migration process that would be great. Thanks!

**Comment by karolkielecki** [comment, marked as answer]
Generally speaking, if you fork and modify the code to your needs, it won't be easy to stay up to date with the next updates and releases. I don't know what custom code you've added to your project. In general, the data migration won't be a problem for migrating from 2.11 to 3.0. How it will work with 2.10 + your custom code, I'm not sure--it depends a lot on what you added. Maybe the 2nd option you mentioned would be better--but I can't guarantee anything.

**Comment by ocelot2123** [comment]
The way we structured our project is we put all our custom code in separate directories and files and not touching any of the saleor files so there shouldn't be any merge conflicts other than a few places. Thanks for the response!
